sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -927,6 +937,179 @@ abstract class GpuBaseAggregateMeta[INPUT <: SparkPlan](
   }
 }
 
+/**
+ * Base class for metadata around `SortAggregateExec` and `ObjectHashAggregateExec`, which may
+ * contain TypedImperativeAggregate functions in aggregate expressions.
+ */
+abstract class GpuNoHashAggregateMeta[INPUT <: SparkPlan](
```

Comment:
Because `GpuSortAggregateMeta` also inherits from it. When `spark.sql.execution.useObjectHashAggregateExec` is set to False, Spark catalyst will plan a `SortAggregateExec` instead of `ObjectAggregateExec` for Aggregate (logical plan) with TypedImperativeAggregate functions.

Reply:
I also think we can come up with a better name, even a long name like. `GpuTypeImperativeSupportedAggregateExecMeta`